src/Account/Response/GetShopCarrierOptionsResponse.php

```diff
@@ -33,20 +33,44 @@ protected function parseResponseBody(): void
     {
         $options = json_decode($this->getBody(), true)['data']['carrier_options'] ?? [];
 
-        $this->options = new CarrierCollection(
-            array_map(static function (array $option) {
-                $contractId = Carrier::TYPE_CUSTOM === $option['type'] ? $option['id'] ?? null : null;
-
-                return [
-                    'id'         => $option['carrier_id'] ?? $option['carrier']['id'] ?? null,
-                    'contractId' => $contractId,
-                    'enabled'    => $option['enabled'] ?? null,
-                    'label'      => $option['label'] ?? null,
-                    'optional'   => $option['optional'] ?? null,
-                    'primary'    => $option['primary'] ?? null,
-                    'type'       => $option['type'] ?? null,
-                ];
-            }, $options)
-        );
+        $collection = new CarrierCollection(array_map(static function (array $option) {
+            $contractId = Carrier::TYPE_CUSTOM === $option['type'] ? $option['id'] ?? null : null;
+
+            return [
+                'id'         => $option['carrier_id'] ?? $option['carrier']['id'] ?? null,
+                'contractId' => $contractId,
+                'enabled'    => $option['enabled'] ?? null,
+                'label'      => $option['label'] ?? null,
+                'optional'   => $option['optional'] ?? null,
+                'primary'    => $option['primary'] ?? null,
+                'type'       => $option['type'] ?? null,
+            ];
+        }, $options));
+
+        $this->options = $this->createCarrierCollection($collection);
+    }
+
+    /**
+     * @param  \MyParcelNL\Pdk\Carrier\Collection\CarrierCollection $collection
+     *
+     * @return \MyParcelNL\Pdk\Carrier\Collection\CarrierCollection
+     */
+    private function createCarrierCollection(CarrierCollection $collection): CarrierCollection
+    {
+        $enabledPostNlCarriers = $collection
+            ->where('id', Carrier::CARRIER_POSTNL_ID)
```

Comment:
Voor de zekerheid, hoeft dit echt alleen voor PostNL? Is het mogelijk om dit nu alvast futureproof te maken voor wanneer andere carriers een custom contract hebben?

Reply:
+1

Reply:
nee dit is echt alleen voor postnl 🥲